Widgets/RoomView.cpp

```diff
@@ -22,49 +22,55 @@ bool InstanceSortFilterProxyModel::lessThan(const QModelIndex &left, const QMode
   return objA->data(Object::kDepthFieldNumber) < objB->data(Object::kDepthFieldNumber);
 }
 
-RoomView::RoomView(QWidget* parent) : AssetView(parent), model(nullptr) {
-  this->SetZoom(1.0);
+RoomView::RoomView(AssetScrollAreaBackground* parent) : AssetView(parent), model(nullptr) {
+  setFixedSize(sizeHint());
   this->sortedInstances = new InstanceSortFilterProxyModel(this);
   this->sortedTiles = new QSortFilterProxyModel(this);
 }
 
 void RoomView::SetResourceModel(ProtoModelPtr model) {
   this->model = model;
-  this->sortedInstances->setSourceModel(model->GetRepeatedSubModel(Room::kInstancesFieldNumber));
-  this->sortedInstances->sort(Room::Instance::kObjectTypeFieldNumber);
-  this->sortedTiles->setSourceModel(model->GetRepeatedSubModel(Room::kTilesFieldNumber));
-  this->sortedTiles->sort(Room::Tile::kDepthFieldNumber);
-  this->SetZoom(1.0);
+
+  if (model != nullptr) {
+    this->sortedInstances->setSourceModel(model->GetRepeatedSubModel(Room::kInstancesFieldNumber));
+    this->sortedInstances->sort(Room::Instance::kObjectTypeFieldNumber);
+    this->sortedTiles->setSourceModel(model->GetRepeatedSubModel(Room::kTilesFieldNumber));
+    this->sortedTiles->sort(Room::Tile::kDepthFieldNumber);
+  }
+  setFixedSize(sizeHint());
+  repaint();
 }
 
 QSize RoomView::sizeHint() const {
   if (!model) return QSize(640, 480);
-  unsigned roomWidth = model->data(Room::kWidthFieldNumber).toUInt(),
-           roomHeight = model->data(Room::kHeightFieldNumber).toUInt();
-  return QSize(roomWidth, roomHeight);
+  QVariant roomWidth = model->data(Room::kWidthFieldNumber), roomHeight = model->data(Room::kHeightFieldNumber);
+  if (!roomWidth.isValid() || !roomHeight.isValid()) return QSize(640, 480);
```

Comment:
Maybe flag this with a TODO(enigma-dev/enigma-dev#1872)

Reply:
(I'd keep the issue links so they're easy to search, although we really lack the infrastructure for this to matter right now. It's possible GitHub may one day support it.)